### PR TITLE
feat: node dmr

### PR DESCRIPTION
**Pull request by YoDaMa**

### CODE REVIEW

File: resolvers/node/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "version": "1.0.0",
+  "description": "Device Model Repository Client for Azure IoT PnP",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Yoseph Maguire",
+  "license": "MIT",
+  "dependencies": {
+    "eslint": "^7.12.0",
```
Comment (rido-min) [inline resolvers/node/package.json:12]: eslint should be under devDependencies. 

there are missing dependencies like `valid-url`

File: resolvers/node/src/modelFetchers.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    const req = https.request(options, res => {
+        debug('statusCode: ', res.statusCode);
+        res.on('data', d => {
+            debug(d);
+        });
+        res.on('end', () => {
+            debug('all data received');
+            break;
```
Comment (rido-min) [inline resolvers/node/src/modelFetchers.js:23]: `break`  produces a SyntaxError

File: resolvers/node/src/resolverClient.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+"use strict"
+
+const { debug } = require('debug')('resolver');
+const https = require('https');
+var validUrl = require('valid-url');
```
Comment (rido-min) [inline resolvers/node/src/resolverClient.js:8]: why do we need valid-url ?

File: resolvers/node/src/resolverClient.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    if (validUrl.isUri(endpoint)) {
+        // endpoint is URL
+        return { 'uri' : endpoint }
+    } else {
+        // endpoint is file
+        return { 'file' : endpoint }
+    }
+}
+
+function dtmiToPath (dtmi) {
```
Comment (rido-min) [inline resolvers/node/src/resolverClient.js:24]: consider use 1 liner like https://github.com/Azure/iot-plugandplay-models/blob/218ba0b5e10e6ee7583f15603a69532a195bcc34/repo-convention.js#L16

File: resolvers/node/src/resolverClient.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        if (!path) {
+            location = url.pathToFileURL(__dirname)
+        } else {
+            location = url.pathToFileURL(path)
+        }
+        const resolver = new ResolverClient(location);
+        return resolver;
+    }
+
+    addEndpoints (endpoints) {
```
Comment (rido-min) [inline resolvers/node/src/resolverClient.js:69]: we dont have the requirements for federation. We dont need to deal with more than one endpoint

File: resolvers/node/src/resolverClient.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        dtmiList.forEach((dtmi) => {
+            // iterate over the DTMIs and then go through the endpoints list to get
+            // the JSON
+            let model;
+            this.endpointList.forEach((repositoryEndpoint) => {
+                // TODO: Implement fetchModel API in ModelFetcher
+                const result = fetchers.fetchModel(repositoryEndpoint);
+                if (result) {
+                    model = result;
+                    break;
```
Comment (rido-min) [inline resolvers/node/src/resolverClient.js:100]: invalid syntax

File: resolvers/node/src/resolverClient.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+"use strict"
+
+const { debug } = require('debug')('resolver');
+const https = require('https');
+var validUrl = require('valid-url');
+const fetchers = require('modelFetchers');
```
Comment (rido-min) [inline resolvers/node/src/resolverClient.js:9]: missing the `./` to indicate the module is local

**Review by rido-min — COMMENTED** [review]
How can we make this library consumable in browsers? (see https://iotmodels.github.io/web-resolver )

Which JS style/flavor we want to use? TS, ES6.. (BTW I'm in love with StandardJS) I see some `;` but not everywhere.

I needed to add a small test script
```JS
const ResolverClient = require('./src/resolverClient')
const rc = new ResolverClient()
rc.resolve('dtmi:com:example:Thermostat;1')
```

File: resolvers/node/README.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# Azure IoT Resolver library for Python
```
Comment (digimaun) [inline resolvers/node/README.md:1]: I agree `Python` is legit ;)

**Pull request closed without merging**